Refacto preprocess

Soso925 opened the pull request.

Soso925 committed:
refacto load(data) by adding paquets

Soso925 committed:
refacto load(data) by adding paquets

Soso925 committed:
add in load a line to consider only top 10 graphememe root

Soso925 committed:
add in load a line to consider only top 10 graphememe root

Soso925 committed:
refacto resize

Soso925 committed:
refacto resize

Soso925 committed:
refacto normalize_img

Soso925 committed:
refacto normalize_img

Soso925 committed:
fix load errors

Soso925 committed:
fix load errors

Soso925 committed:
fix load errors 2

Soso925 committed:
fix load errors 2

Soso925 committed:
create class preprocess_img(starter code)

Soso925 committed:
create class preprocess_img(starter code)

Soso925 committed:
add print info

Soso925 committed:
add print info

Soso925 committed:
mv class to preprocessing.py

Soso925 committed:
mv class to preprocessing.py

Soso925 committed:
fix bug on preprocessing, passing all to numpy array

Soso925 committed:
fix bug on preprocessing, passing all to numpy array

Soso925 committed:
add toto

Soso925 committed:
add toto

Soso925 committed:
minor change in img_id and img last step for load

Soso925 committed:
minor change in img_id and img last step for load

Soso925 committed:
add a global var for RUN_LOCAL, refacto main

Soso925 committed:
add a global var for RUN_LOCAL, refacto main

Alain-TC commented on main.py:
```diff
@@ -3,47 +3,114 @@
+    if RUN_LOCAL == 1: # make a custom model
+        import tensorflow as tf
+         # fix plot_model #@todo : add it in model.py
+         from tensorflow.keras.utils import plot_model
```
Erreur d'indentation

Alain-TC replied:
Faire les imports en début de module

Soso925 replied:
outdated topic

Alain-TC commented on main.py:
```diff
@@ -3,47 +3,114 @@
+         from tensorflow.keras.utils import plot_model
+
+
+        def model_v4(input_h, input_w, summary=False, plot_summary=False):
```
Why definition of the model in main.py?

Soso925 replied:
outdated topic

Soso925 committed:
minor modif on preprocess, work locally

Soso925 committed:
minor modif on preprocess, work locally

Soso925 committed:
everything works here

Soso925 committed:
everything works here

Soso925 committed:
suppress unused functions in utils.py

Soso925 committed:
suppress unused functions in utils.py

Alain-TC commented on .gitignore:
```diff
@@ -4,3 +4,4 @@
 __pycache__/
 data/
 */__pycache__/
+toto
```
to be removed

Soso925 replied:
toto is the output of model history. I think it not necessary to keep it here...

Soso925 replied:
I got it removed. Just wonder why? :)

Alain-TC replied:
nan mais toto c'est pas bon comme nom!

Alain-TC commented on main.py:
```diff
@@ -1,49 +1,33 @@
+from utils import load, preprocess_img, plot_result  # @todo : add other paquet once the training is over
+from preprocessing.preprocessing import one_hot_y, split_train_test
+from models.model import history_save, model_v4
+from include import RUN_LOCAL
```
import inutile

Soso925 replied:
why? I use almost all of the functions above.

Alain-TC commented on models/model.py:
```diff
@@ -2,6 +2,7 @@
 from tensorflow.keras.utils import plot_model
 import pickle
 import json
+from include import RUN_LOCAL
```
import inutilisé

Soso925 replied:
yes, it is actually used in the function model_V4

Alain-TC commented on models/model.py:
```diff
@@ -123,6 +124,7 @@ def model_toto3(summary = False, plot_summary = False) :
 
 
 def model_v4(input_h, input_w, summary=False, plot_summary=False):
+    global RUN_LOCAL
```
De façon générale il faut éviter les variables globales, python est plutôt fonctionnel donc c'est un paradigme différent

Soso925 replied:
Je voulais créer un mode qui permet de tourner le code sur seulement 10 grapheme les plus fréquents. Histoire d'accélérer l'apprentissage. Le fait d'ajouter RUN_LOCAL est un moyen plutôt rapide et safe. Meme si je suis d'accord que le paradigme est différent.

Soso925 replied:
Comme on est sur Colab maintenant, je peux enlever RUN_LOCAL. :)

Alain-TC commented on preprocessing/preprocessing.py:
```diff
@@ -97,6 +90,6 @@ def resize(df, input_h, input_w, size=64,  need_progress_bar=True):
\ No newline at end of file
+            resized.append(resized_roi.reshape(-1))
+    #resized = pd.DataFrame(resized).T
+    return np.array(resized).reshape(-1,resize_size,resize_size)
```
Ajouter une ligne en fin de fichier (PEP8)

Soso925 replied:
Pourquoi?

Alain-TC replied:
C'est les règles de PEP8

Alain-TC commented on utils.py:
```diff
@@ -1,35 +1,70 @@
 
-def load(DATA_FOLDER):
+def load(DATA_FOLDER, paquets = None):
+    global RUN_LOCAL
```
same

Alain-TC commented on utils.py:
```diff
@@ -39,24 +74,44 @@ def sub_plot_res(axes, x, y1, y2, l1='Training gr_accuracy', l2='Dev gr_accuracy
-    val_loss_gr = history.history['val_hgr_loss']
-    val_loss_vd = history.history['val_hvd_loss']
-    val_loss_cd = history.history['val_hcd_loss']
+    try :
```
Pourquoi utiliser un try/error ici?

Soso925 replied:
parce que selon t'es en local ou t'es en bcp, le log est affiché de façon différente. En local, c'est 'hgr_accuracy' alors qu'en Colab c'est 'hgr_acc'.

Soso925 replied:
ps: en Kaggle, c'est aussi 'hgr_accuracy' qui est prise en compte.

Alain-TC replied:
D'où vient la différence? C'est pas les mêmes versions d'un package ? etc?

Soso925 committed:
GLOBAL_RUN Removal + Ajouter une ligne

Soso925 committed:
GLOBAL_RUN Removal + Ajouter une ligne

Soso925 committed:
remove data manip on getting first 1K lines for x and y

Soso925 committed:
remove data manip on getting first 1K lines for x and y

Commit:
Merging
Merge branch 'refacto_preprocess' of https://github.com/Soso925/kaggle_bengaliai_cv19 into refacto_preprocess
